src/System.IO.Compression/tests/DeflateStreamTests.cs

```diff
@@ -1,23 +1,40 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Collections.Generic;
+using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
-using System.IO;
-using System.Reflection;
-using System.IO.Compression;
 using Xunit;
 
 namespace System.IO.Compression.Tests
 {
-    public partial class DeflateStreamTests
+    public class ZLibDeflateStreamTests : DeflateStreamTests, IDisposable
+    {
+        public ZLibDeflateStreamTests() { SetWorkerMode("zlib"); }
+        public void Dispose() { SetWorkerMode("unknown"); }
+    }
+
+    public class ManagedDeflateStreamTests : DeflateStreamTests, IDisposable
+    {
+        public ManagedDeflateStreamTests() { SetWorkerMode("managed"); }
+        public void Dispose() { SetWorkerMode("unknown"); }
+    }
+
+    public abstract class DeflateStreamTests
     {
         static string gzTestFile(String fileName) { return Path.Combine("GZTestData", fileName); }
 
+        internal static void SetWorkerMode(string mode)
```

Comment:
Since this is shared between deflatestream and gzipstream tests, is corefx\src\Common\tests\Compression\Utilities\ a better place?